test/functional/feature_rbf.py

```diff
@@ -115,30 +103,24 @@ def make_utxo(self, node, amount, confirmed=True, scriptPubKey=DUMMY_P2WPKH_SCRI
                 assert new_size < mempool_size
                 mempool_size = new_size
 
-        return COutPoint(int(txid, 16), n)
+        return self.wallet.get_utxo(txid=txid, vout=n)
 
     def test_simple_doublespend(self):
         """Simple doublespend"""
         # we use MiniWallet to create a transaction template with inputs correctly set,
         # and modify the output (amount, scriptPubKey) according to our needs
-        tx_template = self.wallet.create_self_transfer()['tx']
-
-        tx1a = deepcopy(tx_template)
-        tx1a.vout = [CTxOut(1 * COIN, DUMMY_P2WPKH_SCRIPT)]
-        tx1a_hex = tx1a.serialize().hex()
-        tx1a_txid = self.nodes[0].sendrawtransaction(tx1a_hex, 0)
+        tx = self.wallet.create_self_transfer()["tx"]
+        tx1a_txid = self.nodes[0].sendrawtransaction(tx.serialize().hex())
 
         # Should fail because we haven't changed the fee
-        tx1b = deepcopy(tx_template)
-        tx1b.vout = [CTxOut(1 * COIN, DUMMY_2_P2WPKH_SCRIPT)]
-        tx1b_hex = tx1b.serialize().hex()
+        tx.vout[0].scriptPubKey[-1] -= 1
```

Comment:
This would fail if the scriptPubKey ends with a zero byte, e.g.:
```
>>> spk = bytearray(b'\xbe\xef\x00')
>>> spk[-1] -= 1
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: byte must be in range(0, 256)
```
Modifying it with a bit-flip is probably slightly preffered:
```
>>> spk[-1] ^= 1
>>> spk
bytearray(b'\xbe\xef\x01')
```

```suggestion
        tx.vout[0].scriptPubKey[-1] ^= 1
```

Reply:
Thx.

The test should be deterministically pass even with `-=1`, but I've switched to `^=1` as it is cleaner.